iroha/src/sumeragi.rs

```diff
@@ -276,14 +302,66 @@ impl Debug for Sumeragi {
 #[cfg(test)]
 mod tests {
     use super::*;
-    use crate::crypto;
-    use async_std::sync;
+    use crate::{crypto, ToriiMessageReceiver};
+    use async_std::{
+        prelude::*,
+        sync::{self, RwLock},
+    };
+    use std::{collections::HashMap, convert::TryFrom, time::Duration};
+
+    #[derive(Default)]
+    struct Router {
```

Comment:
Please add the description or link to paper

Reply:
It's basically a network simulation through channels. It let's us not depend on torii in tests.

Reply:
The structure is for tests only.